beetmoverscript/tests/test_script.py

```diff
@@ -69,6 +70,45 @@ def fake_list(*args):
         await push_to_releases(context)
 
 
+def test_check_maven_artifact_map(context):
+    context.action = "push-to-maven"
+
+    fake_correct_version = "12.3.20200920201111"
+    source = "fake_path/fake-artifact-{version}.jar"
+
+    def artifact_map_entry_with_version(version):
+        return {
+            "locale": "en-US",
+            "paths": {
+                source: {
+                    "checksums_path": "",
+                    "destinations": [f"fake/destination/{version}/fake-artifact-{version}.jar"],
+                }
+            },
+            "taskId": "fake-task-id",
+        }
+
+    context.task = {
+        "payload": {
+            "artifactMap": [],
+            "releaseProperties": {"appName": "nightly_components"},
+            "upstreamArtifacts": [{"paths": [source], "taskId": "fake-task-id", "taskType": "build"}],
+            "version": "bad.version",
+        }
+    }
+
+    with pytest.raises(ScriptWorkerTaskException):
+        check_maven_artifact_map(context)
+
+    context.task["payload"]["version"] = fake_correct_version
+    context.task["payload"]["artifactMap"] = [artifact_map_entry_with_version(fake_correct_version)]
+    check_maven_artifact_map(context)
+
+    context.task["payload"]["artifactMap"] = [artifact_map_entry_with_version("a.bad.version")]
```

Comment:
Sweet! I love the fact you tested a failure case! 👍 This test looks good the way it is now. If you'd like to enhance it, you can leverage `@pytest.mark.parametrize` https://docs.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising. So this test could be split into 3 different cases. This will make the logic smaller and this will ensure that each case runs even though one fails. For instance: if we break the first `check_maven_artifact_map()` call, then the next ones won't be tested.

That's up to you. I don't mind merging this test as-is.

Reply:
OK. I didn't know you could parametrize when exceptions were involved. I think I'll do what you suggest.